src/stepFunctions/asl/aslServer.ts

```diff
@@ -303,32 +302,34 @@ function validateTextDocument(textDocument: TextDocument, callback?: (diagnostic
 
         return
     }
-    setLanguageService(textDocument)
     const jsonDocument = getJSONDocument(textDocument)
     const version = textDocument.version
 
     const documentSettings: DocumentLanguageSettings = { comments: 'error', trailingCommas: 'error' }
-    languageService.doValidation(textDocument, jsonDocument, documentSettings).then(
-        diagnostics => {
-            setTimeout(() => {
-                const currDocument = documents.get(textDocument.uri)
-                if (currDocument && currDocument.version === version) {
-                    respond(diagnostics) // Send the computed diagnostics to VSCode.
-                }
-            }, 100)
-        },
-        error => {
-            connection.console.error(formatError(`Error while validating ${textDocument.uri}`, error))
-        }
-    )
+    langId = textDocument.languageId
+    getLanguageService()
+        .doValidation(textDocument, jsonDocument, documentSettings)
+        .then(
+            diagnostics => {
+                setTimeout(() => {
+                    const currDocument = documents.get(textDocument.uri)
+                    if (currDocument && currDocument.version === version) {
+                        respond(diagnostics) // Send the computed diagnostics to VSCode.
+                    }
+                }, 100)
+            },
+            error => {
+                connection.console.error(formatError(`Error while validating ${textDocument.uri}`, error))
+            }
+        )
 }
 
 connection.onDidChangeWatchedFiles(change => {
     // Monitored files have changed in VSCode
     let hasChanges = false
     // tslint:disable-next-line: no-unsafe-any
     change.changes.forEach(c => {
-        if (languageService.resetSchema(c.uri)) {
+        if (aslJsonLanguageService.resetSchema(c.uri)) {
```

Comment:
Why would this explicitly be the json service?

Reply:
That's a bug, it should be getLanguageService()